backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/utils/VmDeviceCommonUtils.java

```diff
@@ -237,21 +237,32 @@ private static int setDiskBootOrder(
             Map<VmDeviceId, DiskVmElement> deviceIdTodiskVmElement) {
         LinkedList<VmDevice> diskDevices = new LinkedList<>();
         for (VmDevice device : devices) {
-            if (isDisk(device)) {
-                Guid id = device.getDeviceId();
-                if (id != null && !id.equals(Guid.Empty)) {
-                    if (device.getSnapshotId() == null) {
-                        diskDevices.addFirst(device);
-                    } else {
-                        diskDevices.addLast(device);
-                    }
-                }
+            if (!isDisk(device)) {
+                continue;
             }
+            Guid id = device.getDeviceId();
+            if (id == null || id.equals(Guid.Empty)) {
+                continue;
+            }
+            DiskVmElement dve = deviceIdTodiskVmElement.get(device.getId());
+            if (dve == null) {
+                continue;
+            }
+            if (device.getSnapshotId() == null) {
+                diskDevices.addFirst(device);
+            } else {
+                diskDevices.addLast(device);
+            }
+        }
+
+        if (diskDevices.size() == 1) {
+            diskDevices.get(0).setBootOrder(++bootOrder);
+            return bootOrder;
         }
 
         for (VmDevice device : diskDevices) {
             DiskVmElement dve = deviceIdTodiskVmElement.get(device.getId());
-            if (dve != null && dve.isBoot()) {
+            if (dve.isBoot()) {
```

Comment:
such null checks exist all over the code - I'm not sure if it's because we don't set the diskvmelement for all kinds of disks or just to be on the safe side and prevent theoretical NPE. why dropping the check here? (I see you check this above at line 248..)

Reply:
Because the check above guarantees that `diskDevices` list now contains only devices that have a corresponding `DiskVmElement` in `deviceIdTodiskVmElement` map. And this is important for `diskDevices.size() == 1` check at line 258. So there is no sense to check this on line 265.

Reply:
ah right, missed that somehow